node/src/components/contract_runtime/rewards.rs

```diff
@@ -419,27 +419,36 @@ pub(crate) fn rewards_for_era(
     rewards_info: RewardsInfo,
     current_era_id: EraId,
     core_config: &CoreConfig,
-) -> Result<BTreeMap<PublicKey, U512>, RewardsError> {
+) -> Result<BTreeMap<PublicKey, Vec<U512>>, RewardsError> {
     fn to_ratio_u512(ratio: Ratio<u64>) -> Ratio<U512> {
         Ratio::new(U512::from(*ratio.numer()), U512::from(*ratio.denom()))
     }
 
+    let ratio_u512_zero = Ratio::new(U512::zero(), U512::one());
+    let zero_for_current_era = {
+        let mut map = BTreeMap::new();
+        map.insert(current_era_id, ratio_u512_zero);
+        map
+    };
     let mut full_reward_for_validators: BTreeMap<_, _> = rewards_info
```

Comment:
It's not introduced by the PR, but we seem to be relying on some extra confusing type inference here, because it looks like it's creating a BTreeMap<PublicKey, Ratio<U512>> to me, but it's not and its behavior downstream (as a map of maps) also reflects that. Honestly I'm quite puzzled that this compiles.